### PR TITLE
feat: add boost method

### PR DESCRIPTION
**Pull request by im-adithya**
--

### CODE REVIEW

File: src/utils/keysend.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const TAG_KEYSEND = 'keysend'
+
+export const parseKeysendResponse = (
+  data: Record<string, any>
+): KeysendResponse => {
+  if (data.tag !== TAG_KEYSEND) throw new Error('Invalid keysend params')
+  if (data.status !== "OK") throw new Error('Keysend status not OK')
+
+  if (!('customKey' in data.customData[0]) || data.customData[0]['customKey'] != '696969')
```
Comment (rolznz) [inline src/utils/keysend.ts:11]: for people using their own node they won't have these custom key/value. Should this check be removed?

File: src/lightning-address.ts [hunk 2]
```diff
@@ -104,6 +107,17 @@ export default class LightningAddress {
     return this.generateInvoice(callbackUrl);
   }
 
+  async boost(boost: Boost, amount: number = 0) {
+    const { destination, customKey, customValue } = parseKeysendResponse(this.lnurlpData);
```
Comment (bumi) [inline src/lightning-address.ts:111]: I think this is wrong. this data comes from `this.keysendData`

File: src/lightning-address.ts [hunk 3]
```diff
@@ -104,6 +107,17 @@ export default class LightningAddress {
     return this.generateInvoice(callbackUrl);
   }
 
+  async boost(boost: Boost, amount: number = 0) {
+    const { destination, customKey, customValue } = parseKeysendResponse(this.keysendData);
```
Comment (bumi) [inline src/lightning-address.ts:111]: if we want to validate the keysend resonse. shouldn't we do that maybe better in the `fetch` call?

File: src/lightning-address.ts [hunk 4]
```diff
@@ -47,7 +50,7 @@ export default class LightningAddress {
     const result = await fetch(`${this.options.proxy}?${new URLSearchParams({ ln: this.address }).toString()}`);
     const json = await result.json();
     this.lnurlpData = json.lnurlp;
-    this.keysendData = json.keysend;
+    this.keysendData = parseKeysendResponse(json.keysend);
```
Comment (bumi) [inline src/lightning-address.ts:53]: if we do this here for the keysend params, then lets also do it for the lnurl params

Comment (im-adithya) [reply src/lightning-address.ts:53]: Yes, we are doing it the same way (now that we merged #13)
https://github.com/getAlby/alby-tools/blob/0134d3718efd51652e0a44a2f9b3cfbd13436a93/src/lightning-address.ts#L53
https://github.com/getAlby/alby-tools/blob/0134d3718efd51652e0a44a2f9b3cfbd13436a93/src/lightning-address.ts#L60

**Pull request merged**